### PR TITLE
gh-102839: remove AC for math.log

### PR DESCRIPTION
**Pull request by skirpichev**
Before:
```
$ ./python -m timeit -s 'from math import log' 'log(1.1)' 1000000 loops, best of 5: 344 nsec per loop
$ ./python -m timeit -s 'from math import log' 'log(1.1, 3.2)' 500000 loops, best of 5: 601 nsec per loop
```
After:
```
$ ./python -m timeit -s 'from math import log' 'log(1.1)' 2000000 loops, best of 5: 171 nsec per loop
$ ./python -m timeit -s 'from math import log' 'log(1.1, 3.2)' 1000000 loops, best of 5: 348 nsec per loop
```

* Issue: gh-102839

### CODE REVIEW

File: Modules/mathmodule.c [hunk 1]
```diff
@@ -2284,33 +2284,20 @@ loghelper(PyObject* arg, double (*func)(double))
...
-Return the logarithm of x to the given base.
-
-If the base not specified, returns the natural logarithm (base e) of x.
-[clinic start generated code]*/
-
 static PyObject *
-math_log_impl(PyObject *module, PyObject *x, int group_right_1,
-              PyObject *base)
-/*[clinic end generated code: output=7b5a39e526b73fc9 input=0f62d5726cbfebbd]*/
+math_log(PyObject *module, PyObject * const *args, Py_ssize_t nargs)
```
Comment (sobolevn) [inline Modules/mathmodule.c:2290]: Can you please add a comment about why it is not converted to AC?

File: Misc/NEWS.d/next/Library/2023-03-20-12-21-19.gh-issue-102839.RjRi12.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Improve performance of :func:`math.log` arguments handling.
```
Comment (sobolevn) [inline Misc/NEWS.d/next/Library/2023-03-20-12-21-19.gh-issue-102839.RjRi12.rst]:
```suggestion
Improve performance of :func:`math.log` arguments handling by removing the argument clinic.
```

File: Modules/mathmodule.c (hunk 1 above)
Comment (skirpichev) [reply Modules/mathmodule.c:2290]: Ok, I did.

**Review by corona10 — COMMENTED** [review]
I don't think that this is what we have to go.
See: https://github.com/faster-cpython/ideas/issues/565

**Comment by rhettinger** [comment]
@mdickinson Are you on board with this?

**Comment by mdickinson** [comment]
@rhettinger Yep, fine by me. Looks like a significant performance boost.

**Comment by skirpichev** [comment]
> If and when the AC gets improved, we can convert this back to AC. In the meantime, let's take the performance win.

In fact, it's possible to keep AC stuff, if we "corrupt" it like cmath.log:
```
/*[clinic input]
math.log

    x:    object
    base: object = NULL
    /

log(x[, base=math.e]) -> the logarithm of x to the given base.

If the base not specified, returns the natural logarithm (base e) of x.
[clinic start generated code]*/
```
at cost of slight change in the docstring.  Not sure it worth, this just hide the problem with AC/inspect capabilities.

> My timings are about 7x faster than yours. Are you using a debug build for timings, a non-representative machine, or an older compiler?

@rhettinger, perhaps, the second option.  This was a usual (`./confiqure -q && make -s`) build with gcc Debian(bullseye) 10.2.1-6.

**Pull request merged**